src/index.ts

```diff
@@ -28,18 +31,20 @@ async function launchReScript(watch: boolean): Promise<ReScriptProcess> {
 
   let compileOnce = (_value: unknown) => {};
 
-  function dataListener(chunk: any) {
-    const output = chunk.toString().trimEnd();
-    // eslint-disable-next-line no-console
-    console.log(logPrefix, output);
-    if (watch && output.includes('>>>> Finish compiling')) {
-      compileOnce(true);
+  if (!silent) {
+    function dataListener(chunk: any) {
+      const output = chunk.toString().trimEnd();
+      // eslint-disable-next-line no-console
+      console.log(logPrefix, output);
+      if (watch && output.includes('>>>> Finish compiling')) {
+        compileOnce(true);
+      }
     }
-  }
 
-  const { stdout, stderr } = result;
-  stdout && stdout.on('data', dataListener);
-  stderr && stderr.on('data', dataListener);
+    const { stdout, stderr } = result;
+    stdout && stdout.on('data', dataListener);
+    stderr && stderr.on('data', dataListener);
+  }
 
   if (watch) {
     await new Promise((resolve) => {
```

Comment:
There is a bug here: We shouldn't await for this promise if `silent` is `true`, otherwise, Vite will hang forever.

Reply:
@jihchi Oops... Fixed!